src/app/components/core/manage-services/manage-service-modal/manage-service-modal.component.ts

```diff
@@ -0,0 +1,514 @@
+import { Component, ViewChild, Output, HostListener, EventEmitter } from '@angular/core';
+import { FormBuilder, NgForm } from '@angular/forms';
+import {
+  ProgressBarService, AlertService, ServicesApiService, SchedulesService,
+  ConfigurationService,
+  RolesService,
+  ConfigurationControlService,
+  FileUploaderService
+} from '../../../../services';
+import { DialogService } from '../../../common/confirmation-dialog/dialog.service';
+
+import { AlertDialogComponent } from '../../../common/alert-dialog/alert-dialog.component';
+import { isEmpty, cloneDeep } from 'lodash';
+import { concatMap, delayWhen, retryWhen, take, tap } from 'rxjs/operators';
+import { BehaviorSubject, of, throwError, timer } from 'rxjs';
+import { DocService } from '../../../../services/doc.service';
+import { Router } from '@angular/router';
+import { ConfigurationGroupComponent } from '../../configuration-manager/configuration-group/configuration-group.component';
+import {QUOTATION_VALIDATION_PATTERN} from '../../../../utils';
+
+@Component({
+  selector: 'app-manage-service-modal',
+  templateUrl: './manage-service-modal.component.html',
+  styleUrls: ['./manage-service-modal.component.css']
+})
+export class ManageServiceModalComponent {
+  enabled: Boolean;
+  category: any;
+  isServiceAvailable = false;
+  isServiceEnabled = false;
+  serviceModalName = '';
+  serviceName = '';
+  availableServices = [];
+  packageNamePrefix = 'fledge-service-';
```

Comment:
We can use from expectedServices. Can clean up a bit (for variables), based on that object